integration/README.md

```diff
@@ -0,0 +1,37 @@
+# Integration tests
+
+This directory contains integration tests for the project.
+
+The tree structure generally mirrors the source code tree structure.
+
+Requirements for new files in this directory:
+* Every integration test file **must** use the `integration` build tag
+* Every package **must** be named after its directory with `_integration` appended
+  * Requiring a different package name for integration tests avoids aliasing with the main package.
+
+These requirements are enforced by a unit test in this directory.
+
+## Running integration tests
+
+Integration tests require the following environment variables:
+* `CLOUD_ENV` - set to the cloud environment to use (e.g. `aws`, `azure`, `gcp`)
```

Comment:
1. Why is CLOUD_ENV required rather than optional?
2. Are we going to clean up testutil.GetEnvOrSkipTest(t, "CLOUD_ENV") from every test? Given that we have both directory structure AND build tags, this extra check seems redundant

Reply:
It's a remnant of mixing unit tests and integration tests in the same tree. This check ensured that all integration tests were skipped when running `go test ./...`. With the approach in this PR we can rely on build tags to ignore everything under `integration/` by default.

After this PR is merged we can remove it and derive the cloud from the workspace client.

See: https://github.com/databricks/databricks-sdk-go/blob/6f68afdd47a0c9d85be67e142d5a780ae885b334/config/config.go#L241-L258